app/main/controllers/media/RTMediaLike.php

```diff
@@ -390,7 +389,8 @@ public function before_render() {
 	 * @return string
 	 */
 	public function like_button_filter_nonce( $button ) {
-		$button .= wp_nonce_field( 'rtm_media_like_nonce' . $this->media->id, 'rtm_media_like_nonce', true, false );
+		// This nonce is rendered multiple times on same page, so having common ID will result in HTML warning.
+		$button .= '<input type="hidden" name="rtm_media_like_nonce" value="' . esc_attr( wp_create_nonce( 'rtm_media_like_nonce' . $this->media->id ) ) . '" />';
```

Comment:
Here instead of adding custom HTML can we update ID?

Reply:
There are 2 like buttons for same media, so even if we add ID with media_id appended it'll be the same ID.

Reply:
We can load this nonce one as it is common for both. Means just load once instead of two times. You can add check if it already loaded then don't load the second time. This way no need to add a custom HTML field.

Reply:
Yes we can do it. But is it necessary to give ID? `var like_nonce = $( this ).siblings( 'input[name="rtm_media_like_nonce"]' ).val();` this is being searched in siblings so the performance won't be affected much.

Reply:
And even if the nonce is common, the form for both like buttons is different. If we add common nonce with ID, we'll have to fetch that nonce using JS on form submit (like `var like_nonce = $( this ).siblings( 'ID' ).val();`). In this case if JS is disabled it won't work.

Reply:
Are we using form serialize to submit data?

Reply:
Not form serialization, we're just sending nonce in the form. https://github.com/rtMediaWP/rtMedia/pull/1571/files#diff-178843aed6652ee542e53006028e674aR1786
But if JS is enabled it's handled by ajax otherwise I think it'll be submitted

Reply:
Nope. I don't think for data will submit if JS is disabled. Please check the same scenario and if not happening something like that then we could use `wp_nonce_field` this function once only to load a single field.

Reply:
Form data will be submitted when JS is disabled but won't be able to verify nonce because we're checking for `like_nonce` which is being set in JS only.
So I think the change to make here is to change the nonce name to `like_nonce` from `rtm_media_like_nonce`. But I don't see any way to add only 1 nonce and make it work when JS is disabled.